src/main/proto/spine/options.proto

```diff
@@ -1003,3 +1012,31 @@ message CompareByOption {
     // the lower, enums — from the last number value to the 0th value, etc.
     bool descending = 2;
 }
+
+// Defines the error message used if a `set_once` field is set again.
+//
+// Applies only to the fields marked as `set_once`.
+//
+message IfSetAgainOption {
+
+    // The default error message used when the field value is re-assigned.
+    option (default_message) = "The field `{fieldName}` cannot be re-assigned.";
+
+    // A user-defined error message.
+    //
+    // The specified message may include the following tokens:
+    // 1. `{fieldName}` — the field name.
+    // 2. `{currentValue}` — the current field value.
+    // 3. `{proposedValue}` — the value that was attempted to be set.
+    //
+    // The tokens will be replaced at runtime when the error is constructed.
+    //
+    // Example: Using the `(set_once)` option.
+    //
+    //     message User {
+    //         UserId id = 1 [(set_once) = true,
+    //                        (if_set_again).error_msg = "The student ID field cannot be re-assigned. The current value: `{currentValue}`. The proposed value: `{proposedValue}`."];
```

Comment:
Let's try to shorten this example message:
```
The field `{fieldName}` already has value `{currentValue}`, and cannot be re-assigned to `{proposedValue}`.
```

Reply:
Can we have it as the default message, please? It tells what we need to tell in a general way.

The example here should tell something about the student ID as a particular case of custom domain message.